pkg/kyma/cmd/test/cmd.go

```diff
@@ -0,0 +1,155 @@
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/kyma-incubator/kymactl/internal"
+	kyma_helm "github.com/kyma-incubator/kymactl/internal/helm"
+	"github.com/kyma-incubator/kymactl/internal/step"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/helm/pkg/helm"
+	"k8s.io/helm/pkg/helm/environment"
+	helm_release "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+type Options struct {
+	step.Factory
+	skip []string
+	verbose bool
+}
+
+var testNamespaces = []string{
+	"kyma-system",
+	"istio-system",
+	"knative-serving",
+	"kyma-integration",
+}
+
+var testReleases = map[string]bool{
+	"core": true,
+	"monitoring": false,
+	"logging": false,
+	"istio": true,
+	"knative": true,
+	"application-connector": true,
+}
+
+//NewCmd creates a new install command
+func NewCmd() *cobra.Command {
+	opts := &Options{}
+	cmd := &cobra.Command{
+		Use:   "test",
+		Short: "test kyma installation",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringArrayVar(&opts.skip, "skip", []string{}, "Skip tests for these releases")
+	cmd.Flags().BoolVar(&opts.NonInteractive, "non-interactive", false, "Do not use spinners")
+	cmd.Flags().BoolVarP(&opts.verbose, "verbose", "v", false, "Print additional output")
+	return cmd
+}
+
+func (opts *Options) Run() error {
+	helmConfig := &environment.EnvSettings{TillerConnectionTimeout: 300}
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", clientcmd.RecommendedHomeFile)
+	if err != nil {
+		return err
+	}
+
+	helmClient, err := kyma_helm.New(helmConfig, kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range testNamespaces {
+		err := opts.cleanHelmTestPods(ns)
+		if err != nil {
+			return err
+		}
+	}
+
+	for release := range testReleases {
+		s := opts.NewStep(
+			fmt.Sprintf("Testing release %s", release),
+		)
+		s.Start()
+
+		if opts.skipRelease(helmClient, release) {
+			s.Successf("Skipping release %s", release)
+			continue
+		}
+
+		msg, success, err := opts.testRelease(helmClient, s, release)
+		s.Stop(success)
+		if msg != "" && (!success || opts.verbose) {
+			fmt.Println("--- Log ---")
+			fmt.Print(msg)
+			fmt.Println("---")
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (opts *Options) skipRelease(client helm.Interface, release string) bool {
```

Comment:
Why this is a method defined on the struct instead of just an internal function having an options parameter?

Reply:
Just my nasty habit of having "main" object for a command. It makes it easier to pass parameters and common items around when it is a receiver. I like to make functions which parameters are really the varying part while receiver represents context of the operation (OOP legacy thinking). In this particular example `client helm.Interface` could even be part of receiver to achieve this separation.

TL;DR Because I wrote it so.